druid/src/test/java/org/apache/calcite/test/DruidAdapterIT.java

```diff
@@ -2123,6 +2123,51 @@ public Void apply(ResultSet resultSet) {
     sql(sql).returnsUnordered("C=60").queryContains(druidChecker("'queryType':'timeseries'"));
   }
 
+  /** Tests that the aggregate in the druid query
+   * is of type hyperUnique rather than cardinality */
+  @Test public void testHyperUniqueAggregateProduced() {
+    String sql = "select count(distinct \"user_unique\") as users from \"wiki\"";
+    String aggString = "{'type':'hyperUnique','name':'USERS','fieldName':'user_unique'}";
+    sql(sql, WIKI)
+      .queryContains(druidChecker(aggString));
+  }
+
+  /** Tests that non distinct count aggregates don't use the hyperUnique type
+   * for metrics with type hyperUnique */
+  @Test public void testNonDistinctCountAggregateProducedForHU() {
+    String sql = "select count(\"user_unique\") as users from \"wiki\"";
+    String aggString = "{'type':'count','name':'USERS','fieldName':'user_unique'}";
```

Comment:
I see you've opened CALCITE-1805. I'll remove the test cases that test count in this way for now, and then add the correct tests once CALCITE-1805 is resolved.